// proton-native/bin/utils/createElement.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createElement = createElement;

var _components = require("../components/");

// Creates an element with an element type, props and a root instance
function createElement(type, props) {
  var COMPONENTS = {
    ROOT: function ROOT() {
      return new _components.Root();
    },
    APP: function APP() {
      return new _components.App(props);
    },
    VIEW: function VIEW() {
      return new _components.View(props);
    },
    WINDOW: function WINDOW() {
      return new _components.Window(props);
    },
    VIRTUALTEXT: function VIRTUALTEXT() {
      return new _components.VirtualText(props);
    },
    ROOTTEXT: function ROOTTEXT() {
      return new _components.RootText(props);
    },
    IMAGE: function IMAGE() {
      return new _components.Image(props);
    },
    TEXTINPUT: function TEXTINPUT() {
      return new _components.TextInput(props);
    },
    PICKERINTERNAL: function PICKERINTERNAL() {
      return new _components.PickerInternal(props);
    },
    "default": undefined
  };
  return COMPONENTS[type]() || COMPONENTS["default"];
}
