// proton-native/bin/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AppRegistry", {
  enumerable: true,
  get: function get() {
    return _render["default"];
  }
});
Object.defineProperty(exports, "Text", {
  enumerable: true,
  get: function get() {
    return _reactComponents.Text;
  }
});
Object.defineProperty(exports, "TouchableWithoutFeedback", {
  enumerable: true,
  get: function get() {
    return _reactComponents.TouchableWithoutFeedback;
  }
});
Object.defineProperty(exports, "TouchableOpacity", {
  enumerable: true,
  get: function get() {
    return _reactComponents.TouchableOpacity;
  }
});
Object.defineProperty(exports, "TouchableHighlight", {
  enumerable: true,
  get: function get() {
    return _reactComponents.TouchableHighlight;
  }
});
Object.defineProperty(exports, "Button", {
  enumerable: true,
  get: function get() {
    return _reactComponents.Button;
  }
});
Object.defineProperty(exports, "Picker", {
  enumerable: true,
  get: function get() {
    return _reactComponents.Picker;
  }
});
Object.defineProperty(exports, "StyleSheet", {
  enumerable: true,
  get: function get() {
    return _misc.StyleSheet;
  }
});
exports.PickerInternal = exports.TextInput = exports.Image = exports.VirtualText = exports.RootText = exports.App = exports.View = exports.Window = void 0;

var _render = _interopRequireDefault(require("./render"));

var _reactComponents = require("./react-components");

var _misc = require("./misc");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

require('./utils/requireImpl'); // Aliases for createElement method


var Window = 'WINDOW';
exports.Window = Window;
var View = 'VIEW';
exports.View = View;
var App = 'APP';
exports.App = App;
var VirtualText = 'VIRTUALTEXT';
exports.VirtualText = VirtualText;
var RootText = 'ROOTTEXT';
exports.RootText = RootText;
var Image = 'IMAGE';
exports.Image = Image;
var TextInput = 'TEXTINPUT';
exports.TextInput = TextInput;
var PickerInternal = 'PICKERINTERNAL';
exports.PickerInternal = PickerInternal;
